FilmFestivalLoader/IFFR/parse_iffr_html.py

```diff
@@ -547,10 +461,143 @@ def handle_data(self, data):
+        if len(urls) > 0:
+            program = self.iffr_data.get_film_by_key(None, urls[0])
+            screenings = [s for s in self.iffr_data.screenings if s.film.filmid == self.film.filmid]
+            for screening in screenings:
+                screening.combination_program = program
+            if self.audience == 'publiek':
+                print(f'--  combination:{program}')
+                print("---SCREENINGS UPDATED")
+                print()
+
+    def handle_starttag(self, tag, attrs):
+        HtmlPageParser.handle_starttag(self, tag, attrs)
+
+        if self.stateStack.state_is(self.CombinationsParseState.IDLE) and tag == 'h3' and len(attrs) > 0:
+            attr = attrs[0]
+            if attr[0] == 'class' and attr[1] == 'sc-crzoAE hwJoPF':
+                self.stateStack.push(self.CombinationsParseState.AWAITING_SCREENED_FILMS)
+                self.print_debug('Entering SCREENED FILMS section', f'{self.film.title}')
+        elif self.stateStack.state_is(self.CombinationsParseState.IN_SCREENED_FILMS) and tag == 'article':
+            self.stateStack.push(self.CombinationsParseState.IN_SCREENED_FILM)
+        elif self.stateStack.state_is(self.CombinationsParseState.IN_SCREENED_FILM):
+            if tag == 'a' and len(attrs) > 1:
+                attr = attrs[1]
+                if attr[0] == 'href':
+                    self.screened_url = f'{iffr_hostname}{attr[1]}'
+                    self.print_debug('SCREENED URL FOUND', f'{self.screened_url}')
+                    self.stateStack.push(self.CombinationsParseState.IN_SCREENED_URL)
+        elif self.stateStack.state_is(self.CombinationsParseState.IN_SCREENED_URL) and tag == 'h4':
+            if attrs[0][1].endswith('tile__title'):
+                self.stateStack.change(self.CombinationsParseState.IN_SCREENED_TITLE)
+        elif self.stateStack.state_is(self.CombinationsParseState.AWAITING_SCREENED_DESCRIPTION) and tag == 'p':
+            self.stateStack.change(self.CombinationsParseState.IN_SCREENED_DESCRIPTION)
+
+    def handle_endtag(self, tag):
+        HtmlPageParser.handle_endtag(self, tag)
+
+        if self.stateStack.state_is(self.CombinationsParseState.IN_SCREENED_FILM) and tag == 'article':
+            self.stateStack.pop()
+            self.add_screened_film()
+
+    def handle_data(self, data):
+        HtmlPageParser.handle_data(self, data)
+
+        if self.stateStack.state_is(self.CombinationsParseState.AWAITING_SCREENED_FILMS):
+            if data == 'In dit verzamelprogramma':
+                self.stateStack.change(self.CombinationsParseState.IN_SCREENED_FILMS)
+            else:
+                self.stateStack.change(self.CombinationsParseState.DONE)
+        elif self.stateStack.state_is(self.CombinationsParseState.IN_SCREENED_FILMS):
+            if data.startswith('Programma IFFR'):
+                self.stateStack.pop()
+                self.stateStack.change(self.CombinationsParseState.DONE)
```

Comment:
Please consider to use one less stack layer as follows:
- IDLE→ IN_SCREENED_FIMS ↑ IN_SCREENED_FILM →.. → ... ↓ IN_SCREENED_FILMS → DONE

Instead of:
- IDLE ↑ IN_SCREENED_FIMS ↑ IN_SCREENED_FILM →.. → ... ↓ IN_SCREENED_FILMS ↓ IDLE → DONE

Reply:
Done.